common/static/common/js/discussion/views/discussion_thread_list_view.js

```diff
@@ -493,29 +511,126 @@
+                var $inputText = $('.forum-nav-browse-filter-input'),
+                    $filteredMenuItems = $('.forum-nav-browse-menu-item:visible'),
+                    filteredMenuItemsLen = $filteredMenuItems.length,
+                    $curOption = $filteredMenuItems.eq(0).find('.forum-nav-browse-title').eq(0),
+                    $activeOption, $prev, $next;
+
+                switch (event.keyCode) {
+                case 13: {  // enter key
+                    $activeOption = $filteredMenuItems.find('.forum-nav-browse-title.is-focused');
+                    if ($inputText.val() !== '' && $inputText.focus()) {
+                        $activeOption.trigger('click');
+                        this.filterInputReset();
+                    }
+                    break;
+                }
+
+                case 27: { // escape key
+                    this.toggleBrowseMenu(event);
+                    $('.forum-nav-browse-filter-input').val('');
+                    this.filterInputReset();
+                    $('.all-topics').trigger('click');
+                    break;
+                }
+
+                case 38: { // up arrow
+                    if (this.selectedTopicIndex > -1) {
+                        $prev = $('.forum-nav-browse-menu-item:visible')
+                            .eq(this.selectedTopicIndex).find('.forum-nav-browse-title')
+                            .eq(0);
+                        this.selectedTopicIndex -= 1;
+                        if (this.isBrowseMenuVisible()) {
+                            this.filterEnabled = false;
+                            $curOption.removeClass('is-focused');
+                            $prev.addClass('is-focused');
+                            this.selectOption($prev);
+                        } else {
+                            this.selectOption($prev);
+                        }
+                    }
+                    break;
+                }
+
+                case 40: { // down arrow
+                    if (this.selectedTopicIndex < filteredMenuItemsLen - 1) {
+                        this.selectedTopicIndex += 1;
+                        if (this.isBrowseMenuVisible()) {
+                            $next = $('.forum-nav-browse-menu-item:visible')
+                                .eq(this.selectedTopicIndex).find('.forum-nav-browse-title')
+                                .eq(0);
+                            this.filterEnabled = false;
+                            $curOption.removeClass('is-focused');
+                            $next.addClass('is-focused');
+                            this.selectOption($next);
+                        } else {
+                            this.selectOption($next);
+                        }
+                    }
+                    break;
+                }
+
+                default: {
+                    break;
+                }
+
+                }
+                return true;
```

Comment:
What was the eslint error? I'm just curious, seems like the return is not needed so it could be we need to tweak the rule.